pkgs/coverage/lib/src/collect.dart

```diff
@@ -103,6 +106,13 @@ Future<Map<String, dynamic>> collect(Uri serviceUri, bool resume,
         isolateIds,
         coverableLineCache,
         waitPaused);
+
+    // Apply filtering if a filter function is provided
+    if (filter != null) {
```

Comment:
What are these changes for? They seem unrelated to the PR. There's already a bunch of ways of filtering the coverage, so I'm not sure what this is for.

If you check what the `_getAllCoverage` function actually returns, this filter definitely won't work as you intend. Did you test your code?